scripts/bfasst/compare_waveforms/tools/run_vivado.py

```diff
@@ -1,15 +1,27 @@
 """A launcher for Vivado to compare gtkwave wavefiles to Vivado's waveform analysis"""
+import argparse
 from pathlib import Path
-from subprocess import Popen
+import subprocess
 import shutil
-import sys
 
-def main():
-    """The main function."""
-    assert sys.argv[7] is not None, "VIVADO_PATH environmental variable was not set!"
+def parse_args():
+
+    """Creates the argument parser for the Vivado Launcher."""
+
+    parser = argparse.ArgumentParser(description="Launch Vivado.")
+    parser.add_argument("base", metavar="Base", help="Base path to files.")
+    parser.add_argument("module", metavar="Module", help="Module name.")
+    parser.add_argument("temp", metavar="Temp Folder", help=
+    "Temporary Location for TCL files.")
+    parser.add_argument("vivado", metavar="Vivado", help="Location of Vivado Launcher.")
+
+    global args
+    args = parser.parse_args()
```

Comment:
Just a thought, if you `return parser.parse_args()`, then you don't need the global, you can receive the result in main.